File: bot-api/java/src/main/java/dev/robocode/tankroyale/botapi/events/CustomEvent.java
package dev.robocode.tankroyale.botapi.events;

/**
 * A custom event occurring when a condition has been met.
 */
public final class CustomEvent extends BotEvent {

    // Condition that was met to trigger this custom event.
    private final Condition condition;

    /**
     * Initializes a new instance of the CustomEvent class.
     *
     * @param turnNumber is the turn number when the condition was met.
     * @param condition  is the condition that has been met.
     */
    public CustomEvent(int turnNumber, Condition condition) {
        super(turnNumber);
        this.condition = condition;
    }

    /**
     * Returns the condition that was met to trigger this custom event.
     *
     * @return The condition that was met to trigger this custom event.
     */
    public Condition getCondition() {
        return condition;
    }
}

File: bot-api/python/src/tank_royale/bot_api/events/custom_event.py
from tank_royale.bot_api.events.bot_event import BotEvent
from tank_royale.bot_api.events.condition import Condition


class CustomEvent(BotEvent):
    """A custom event occurring when a condition has been met."""

    def __init__(self, turn_number: int, condition: Condition):
        """Initializes a new instance of the CustomEvent class.

        Args:
            turn_number: The turn number when the condition was met.
            condition: The condition that has been met.
        """
        super().__init__(turn_number)
        self.condition = condition

    def get_condition(self) -> Condition:
        """Returns the condition that was met to trigger this custom event.

        Returns:
            The condition that was met to trigger this custom event.
        """
        return self.condition